app/src/main/java/io/qtalk/qgamejsinterfacetester/JSInterface.kt

```diff
@@ -0,0 +1,71 @@
+package io.qtalk.qgamejsinterfacetester
+
+import android.webkit.JavascriptInterface
+
+@Suppress("unused")
+class JSInterface(private var jsInterfaceBridge: JSInterfaceBridge? = null) {
+
+
+    fun setJSInterfaceBridge(jsInterfaceBridge: JSInterfaceBridge?){
+        this.jsInterfaceBridge = jsInterfaceBridge
+    }
+
+    @JavascriptInterface
+    fun getUserAuthToken(): String {
+        return jsInterfaceBridge?.getUserAuthToken()?:""
+    }
+
+    @JavascriptInterface
+    fun notifyGameRoundStarted(){
+        jsInterfaceBridge?.notifyGameRoundStarted()
+    }
+
+    @JavascriptInterface
+    fun notifyGameRoundEnded(){
+        jsInterfaceBridge?.notifyGameRoundEnded()
+    }
+
+    @JavascriptInterface
+    fun updateGamePrompts(prompts: String){
+        jsInterfaceBridge?.updateGamePrompts(prompts)
+    }
+
+    @JavascriptInterface
+    fun clearGamePrompts() {
+        jsInterfaceBridge?.clearGamePrompts()
+    }
+
+    @Deprecated("No supported any more", replaceWith = ReplaceWith("saveBase64Image(base64EncodedImageString)"))
+    @JavascriptInterface
+    fun onBase64ImageSaved(base64EncodedImageString: String){
+        saveBase64Image(base64EncodedImageString)
+    }
+
+    @JavascriptInterface
+    fun saveBase64Image(base64EncodedImageString: String){
+        jsInterfaceBridge?.saveBase64Image(base64EncodedImageString)
+    }
+
+    // test only
+    @JavascriptInterface
+    fun clearWebViewCache(){
+        jsInterfaceBridge?.clearWebViewCache()
+    }
+
+    interface JSInterfaceBridge{
+
+        fun getUserAuthToken(): String
+
+        fun notifyGameRoundStarted()
+
+        fun notifyGameRoundEnded()
+
+        fun updateGamePrompts(prompts: String)
+
+        fun clearGamePrompts()
+
+        fun saveBase64Image(base64EncodedImageString: String)
+
+        fun clearWebViewCache()
+    }
+}
```

Comment:
All IDEs when generating a new file, leave a new line at the end, this is because some programs and systems fail when they do not have a new line in the end as the last line is ignore due to inclusivity and exclusivity rules. Check this answer, https://stackoverflow.com/a/729795/4820556 

Resolution, leave a new line here.

Reply:
done